hosted-cluster-config-operator/controllers/resources/resources.go

```diff
@@ -104,8 +157,179 @@ func (r *reconciler) reconcile(ctx context.Context) error {
+		errs = append(errs, fmt.Errorf("failed to reconcile the kube apiserver service monitor: %w", err))
+	}
+
+	return errors.NewAggregate(errs)
+}
+
+func (r *reconciler) reconcileCRDs(ctx context.Context) error {
+	var errs []error
+
+	requestCount := manifests.RequestCountCRD()
+	if _, err := r.CreateOrUpdate(ctx, r.client, requestCount, func() error {
+		return crd.ReconcileRequestCountCRD(requestCount)
+	}); err != nil {
+		errs = append(errs, fmt.Errorf("failed to reconcile request count crd: %w", err))
+	}
+
+	return errors.NewAggregate(errs)
+}
+
+func (r *reconciler) reconcileConfig(ctx context.Context, hcp *hyperv1.HostedControlPlane, globalConfig globalconfig.GlobalConfig) error {
+	var errs []error
+
+	apiServerAddress := hcp.Status.ControlPlaneEndpoint.Host
+
+	if len(apiServerAddress) == 0 {
+		return fmt.Errorf("hosted control plane does not have an APIServer endpoint address")
+	}
+
+	// Infrastructure is first reconciled for its spec
+	infra := globalconfig.InfrastructureConfig()
+	if _, err := r.CreateOrUpdate(ctx, r.client, infra, func() error {
+		globalconfig.ReconcileInfrastructure(infra, hcp)
+		return nil
+	}); err != nil {
+		errs = append(errs, fmt.Errorf("failed to reconcile infrastructure config spec: %w", err))
+	}
+	// It is reconciled a second time to update its status
+	globalconfig.ReconcileInfrastructure(infra, hcp)
+	if err := r.client.Status().Update(ctx, infra); err != nil {
+		errs = append(errs, fmt.Errorf("failed to update infrastructure status: %w", err))
+	}
+
+	dns := globalconfig.DNSConfig()
+	if _, err := r.CreateOrUpdate(ctx, r.client, dns, func() error {
+		globalconfig.ReconcileDNSConfig(dns, hcp)
+		return nil
+	}); err != nil {
+		errs = append(errs, fmt.Errorf("failed to reconcile dns config: %w", err))
+	}
+
+	ingress := globalconfig.IngressConfig()
+	if _, err := r.CreateOrUpdate(ctx, r.client, ingress, func() error {
+		globalconfig.ReconcileIngressConfig(ingress, hcp, globalConfig)
+		return nil
+	}); err != nil {
+		errs = append(errs, fmt.Errorf("failed to reconcile ingress config: %w", err))
+	}
+
+	network := globalconfig.NetworkConfig()
+	if _, err := r.CreateOrUpdate(ctx, r.client, network, func() error {
+		globalconfig.ReconcileNetworkConfig(network, hcp, globalConfig)
+		return nil
+	}); err != nil {
+		errs = append(errs, fmt.Errorf("failed to reconcile network config: %w", err))
+	}
+
+	proxy := globalconfig.ProxyConfig()
+	if _, err := r.CreateOrUpdate(ctx, r.client, proxy, func() error {
+		globalconfig.ReconcileProxyConfig(proxy, hcp, globalConfig)
+		return nil
+	}); err != nil {
+		errs = append(errs, fmt.Errorf("failed to reconcile proxy config: %w", err))
+	}
+
+	installConfigCM := manifests.InstallConfigConfigMap()
+	if _, err := r.CreateOrUpdate(ctx, r.client, installConfigCM, func() error {
+		if installConfigCM.Data == nil {
+			installConfigCM.Data = map[string]string{}
+		}
+		installConfigCM.Data["install-config"] = globalconfig.NewInstallConfig(hcp).String()
```

Comment:
Do we want to keep additional keys if any?

Reply:
no, will fix